x/incentives/keeper/incentives.go

```diff
@@ -0,0 +1,113 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/tharsis/evmos/x/incentives/types"
+)
+
+// GetAllIncentives - get all registered Incentives
+func (k Keeper) GetAllIncentives(ctx sdk.Context) []types.Incentive {
+	incentives := []types.Incentive{}
+
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.KeyPrefixIncentive)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var incentive types.Incentive
+		k.cdc.MustUnmarshal(iterator.Value(), &incentive)
+
+		incentives = append(incentives, incentive)
+	}
+
+	return incentives
+}
+
+// IterateIncentives iterates over all registered `Incentives` and performs a
+// callback.
+func (k Keeper) IterateIncentives(
+	ctx sdk.Context,
+	handlerFn func(incentive types.Incentive) (stop bool),
+) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.KeyPrefixIncentive)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var incentive types.Incentive
+		k.cdc.MustUnmarshal(iterator.Value(), &incentive)
+
+		if handlerFn(incentive) {
+			break
+		}
+	}
+}
+
+// GetIncentive - get registered incentive from the identifier
+func (k Keeper) GetIncentive(
+	ctx sdk.Context,
+	contract common.Address,
+) (types.Incentive, bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixIncentive)
+	var incentive types.Incentive
+	bz := store.Get(contract.Bytes())
+	if len(bz) == 0 {
+		return types.Incentive{}, false
+	}
+
```

Comment:
nit, we don't need to assign the variable until we unmarshal
```suggestion
	bz := store.Get(contract.Bytes())
	if len(bz) == 0 {
		return types.Incentive{}, false
	}

	var incentive types.Incentive
```